internal/resources/grafana/resource_report.go

```diff
@@ -571,3 +580,40 @@ func parseCustomReportInterval(i interface{}) (int, string, error) {
 
 	return number, unit, nil
 }
+
+func validateReportVariables(i interface{}, path cty.Path) diag.Diagnostics {
+	m, ok := i.(map[string]interface{})
+	if !ok {
+		return diag.FromErr(errors.New("report_variables schema should be a map of strings separated by commas"))
+	}
+
+	for _, v := range m {
+		if _, ok := v.(string); !ok {
+			return diag.FromErr(fmt.Errorf("value %#v isn't a string", v))
+		}
+	}
+
+	return nil
+}
+
+func parseReportVariables(reportVariables interface{}) map[string]interface{} {
+	if reportVariables == nil {
+		return nil
+	}
+	rvMap := reportVariables.(map[string]interface{})
+	newMap := make(map[string]interface{}, len(rvMap))
+	for k, rv := range rvMap {
+		switch rvType := rv.(type) {
+		case []interface{}:
+			values := make([]string, len(rvType))
+			for i, v := range rvType {
+				values[i] = v.(string)
+			}
+			newMap[k] = strings.Join(values, ",")
+		case string:
+			newMap[k] = strings.Split(rvType, ",")
+		}
+	}
+
+	return newMap
+}
```

Comment:
Took me a while to understand that this function goes both ways.

This is a bit of a nit, but even if it's more code, I'd still rather have two functions (pack/unpack) with type validation in each (and errors) than a single function that converts depending on the input

Reply:
I split it into two functions. One for the request and the other for the response. 

In for request `interface{}` is the strings separated by coma and for the response, the interface{} is returned as an array of interfaces (a list of string).

Reply:
Thank you!